samza-core/src/main/java/org/apache/samza/serializers/model/JsonJobModelMixIn.java

```diff
@@ -30,12 +30,12 @@
  */
 public abstract class JsonJobModelMixIn {
   @JsonCreator
-  public JsonJobModelMixIn(@JsonProperty("config") Config config, @JsonProperty("containers") Map<Integer, ContainerModel> containers) {
+  public JsonJobModelMixIn(@JsonProperty("config") Config config, @JsonProperty("containers") Map<String, ContainerModel> containers) {
```

Comment:
I verified the behavior of json parser in jackson library. Unless we configure json parser to "ALLOW_UNQUOTED_FIELD_NAMES", it will always serialize the field name as String and it will correctly deserialize it back to Integer/String as specified in the MixinAnnotation. 

For existing jobs, we can safely assume that containerId/processorId is always an integer value. So, it shouldn't be a problem to rollback or roll-forward.